pkg/provisioner/ironic/ironic.go

```diff
@@ -249,35 +249,37 @@ func (p *ironicProvisioner) findExistingHost(bootMACAddress string) (ironicNode
 		}
 	}
 
-	// Try to load the node by port address
-	p.log.Info("looking for existing node by MAC", "MAC", bootMACAddress)
-	allPorts, err := p.listAllPorts(bootMACAddress)
+	// If mac address is present in host config, try to load the node by this port with mac address.
+	if bootMACAddress != "" {
```

Comment:
nit: an early return here would be easier to read.

Reply:
a good coding tip 😃 . From the `findExistingHost` function, it supports multiple methods to find an existing node, and other methods may be added in the future, **the early return is not very good at showing the order in which all methods were tried**. Therefore, it is recommended to use the method to find an node by port **only** when the value of bootMACAddress is present.